docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,399 @@
+# CCv1 Image Security Design
+
+This document describes the design of container image security under the confidential container V1 architecture, including the encryption-decryption and signing of container image. The components involved include image-rs, ocicrypt-rs, attestation-agent and Key Broker Service.
+
+## Production of protected container image
+
+In the confidential containers V1 design, after the user completes the production of the conventional container image, it is necessary to add security protection functions for the container image, including encryption and signing. It should be noted that the encryption process must be carried out before the signing process.
```

Comment:
@jodh-intel I believe that you are asking good questions. The "atomically" part is more a question of integrity than anything else, right? Can we check the encrypted/signed image against the original file at the end of the process to ensure that nothing change in between? Is your concern about "modify the image" regarding the unencrypted source, or something else?

Also, out of curiosity, what kind of attack do you have in mind when you write "different physical system to the one where the image is to be deployed"?

Reply:
Terminology (here and in the rest of the document): Should we use "owner" instead of "user"? (I asked on Slack)

Reply:
On that topic, CC _really_ needs a glossary folks: https://github.com/confidential-containers/documentation/issues/13.

Reply:
@c3d - I'm just trying to understand the security doc in the context of the CC threat model. CC does not trust the host environment so if the newly created, but unencrypted and unsigned image is left lying around in that untrusted host environment, how can you then trust the resulting encrypted and signed image, since it may have been compromised before you actually encrypted and signed it? To counter these attack vectors, presumably you would want to as a minimum only ever create the secure images on a fully trusted host. But for the extra paranoid, you would want to create the image, then immediately encrypt and sign it (and verify that it can be decrypted and its signature is valid of course).

Reply:
@jodh-intel wrote:
>To counter these attack vectors, presumably you would want to as a minimum only ever create the secure images on a fully trusted host

Ah, now I get your point. But you could be in the same host in a different VM you own, right? So what you really mean is that you want the image to be encrypted in a owner-trusted environment, and that this should not be done on a system that cannot be considered trusted, including the host itself.